activerecord/lib/active_record/connection_adapters/postgresql/oid/hstore.rb

```diff
@@ -1,26 +1,71 @@
 # frozen_string_literal: true
 
+require "strscan"
+
 module ActiveRecord
   module ConnectionAdapters
     module PostgreSQL
       module OID # :nodoc:
         class Hstore < Type::Value # :nodoc:
+          ERROR = "Invalid Hstore document: %s"
+
           include ActiveModel::Type::Helpers::Mutable
 
           def type
             :hstore
           end
 
           def deserialize(value)
-            if value.is_a?(::String)
-              ::Hash[value.scan(HstorePair).map { |k, v|
```

Comment:
doesn't look like `HstorePair` is referenced anywhere anymore. could it be removed?

Reply:
I was in another commit on master.

Reply:
oops just saw the same comment above